mautic/mautic
# M3: Plugin bundle upgrade

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8007
| BC breaks? | Y
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Upgrade plugin bundle to support S3 and remove deprecations.

#### Steps to test this PR:
1. Test forms with SF3

#### List backwards compatibility breaks:
1. Removed deprecated method Mautic\PluginBundle\Controller\AjaxController::getIntegrationLeadFieldsAction()
2. Removed deprecated method Mautic\PluginBundle\Controller\AjaxController::getIntegrationCompanyFieldsAction()
3. Removed deprecated method Mautic\PluginBundle\Integration\AbstractIntegration::init()
4. Removed deprecated method Mautic\PluginBundle\Integration\AbstractIntegration::getUserId()
5. Changed constructor args in \Mautic\PluginBundle\Integration\AbstractIntegration and removed setter methods used by IntegrationPass
6. Removed \CoreBundle\DependencyInjection\Compiler\IntegrationPass
7. Removed \PluginBundle\Controller\AjaxController::getIntegrationLeadFieldsAction()
8. Removed \PluginBundle\Controller\AjaxController::getIntegrationCompanyFieldsAction()
9. Removed constructor from \Mautic\PluginBundle\Form\Type\FieldsType
10. Removed constructor from \Mautic\PluginBundle\Form\Type\CompanyFieldsType

#### Leaved as it is:
 - `\Mautic\PluginBundle\Helper\EventHelper::pushLead()`
It is called from `\Mautic\PointBundle\Model\TriggerModel::triggerEvents` which is part of PluginBundle.
 -  `\Mautic\PluginBundle\Bundle\PluginBundleBase` 
This needs to be separated Issue/PR I think.

#### Testing steps:
- Test plugin page. You have to see same plugin count.
- Test plugin installation/update/reload via command.
- Test Onesignal integration config
- Test Outlook integration config
- Test Facebook integration config
- Test Foursquare integration config
- Test Googleplus integration config
- Test Instagram integration config
- Test Linkedin integration config
- Test Twitter integration config
- Test configuration of linkedin plugin
